tag/queries_test.go

```diff
@@ -0,0 +1,93 @@
+package tag_test
+
+import (
+	"testing"
+
+	"github.com/jinzhu/gorm"
+	"github.com/traggo/server/schema"
+	"github.com/traggo/server/tag"
+	"github.com/traggo/server/test"
+)
+
+func Test_noTags_get(t *testing.T) {
+	query := `
+query {
+   tags {
+      key,
+      color,
+      type
+   }
+}`
+	expected := `
+{
+  "tags": []
+}`
+	test.NewGQL(t).Queries(tag.Queries).Exec(query).Succeeds(expected)
+}
+
+func Test_existingTags_get(t *testing.T) {
+	query := `
+query {
+   tags {
+      key,
+      color,
+      type
+   }
+}`
+	createTags := func(db *gorm.DB) {
+		db.Create(&schema.TagDefinition{Key: "my tag", Color: "#fff", Type: schema.TypeSingleValue})
+		db.Create(&schema.TagDefinition{Key: "my tag 2", Color: "#fff", Type: schema.TypeSingleValue})
+	}
+
+	expected := `
+{
+  "tags": [
+    {
+      "color": "#fff",
+      "key": "my tag",
+      "type": "singlevalue"
+    },
+    {
+      "color": "#fff",
+      "key": "my tag 2",
+      "type": "singlevalue"
+    }
+  ]
+}`
+	test.NewGQL(t).Queries(tag.Queries).BeforeExec(createTags).Exec(query).Succeeds(expected)
+
+}
+
+func Test_suggestTag_get(t *testing.T) {
+	query := `
+query {
+   suggestTag(query: "pr") {
+      key,
+      color,
+      type
+   }
+}`
+	createTags := func(db *gorm.DB) {
+		db.Create(&schema.TagDefinition{Key: "project", Color: "#fff", Type: schema.TypeSingleValue})
+		db.Create(&schema.TagDefinition{Key: "priority", Color: "#fff", Type: schema.TypeSingleValue})
+		db.Create(&schema.TagDefinition{Key: "wood", Color: "#fff", Type: schema.TypeSingleValue})
+	}
+
+	expected := `
+{
+  "suggestTag": [
+    {
+      "color": "#fff",
+      "key": "project",
+      "type": "singlevalue"
+    },
+    {
+      "color": "#fff",
+      "key": "priority",
+      "type": "singlevalue"
+    }
+  ]
+}`
+	test.NewGQL(t).Queries(tag.Queries).BeforeExec(createTags).Exec(query).Succeeds(expected)
+
+}
```

Comment:
In https://github.com/traggo/server/pull/3/files#diff-c8f7c60b36faa33b80e854e7de49b103 you kind of had the test naming format `Test_unitToTest_result|circumstances`. Here you have `Test_result|circumstances_unitToTest`.
Is there a reason?

Reply:
Should there be a test like `Test_missingTag_get`, where you put a tag but search for another?
With this you would ensure that `get` is not just giving back "random" results.

Reply:
Good idea! I'll do this.